File: example/ruby_client/two_legged_oauth.rb
#!/usr/bin/env ruby

require 'rubygems'
require 'net/http'
require 'oauth'

consumer = OAuth::Consumer.new('CONSUMER_KEY', 'CONSUMER_SECRET')

# GET expected 401
http = Net::HTTP.new('localhost', 8080)
req = Net::HTTP::Get.new('/')
res = http.request(req)
puts res.code
puts res.body

# GET expected 200
http = Net::HTTP.new('localhost', 8080)
req = Net::HTTP::Get.new('/2legged/')
req.oauth!(http, consumer)
res = http.request(req)
puts res.code
puts res.body

# POST expected 401
req = Net::HTTP::Post.new('/2legged/post/submit')
req.set_form_data({:id => :foo, :password => :bar})
res = http.request(req)
puts res.code
puts res.body


# POST expected 400
req = Net::HTTP::Post.new('/2legged/post/submit')
req.set_form_data({:id => :foo})
req.oauth!(http, consumer)
res = http.request(req)
puts res.code
puts res.body

# POST expected 200
req = Net::HTTP::Post.new('/2legged/post/submit')
req.set_form_data({:id => :foo, :password => :bar})
req.oauth!(http, consumer)
res = http.request(req)
puts res.code
puts res.body

File: example/perl_client/two_legged_oauth.pl
#!/usr/bin/perl

use strict;
use warnings;

use Data::Dumper;
use YAML;
use OAuth::Lite;
use OAuth::Lite::Consumer;

my $oauth = OAuth::Lite::Consumer->new(
 consumer_key => 'CONSUMER_KEY',
 consumer_secret => 'CONSUMER_SECRET',
);
print YAML::Dump($oauth->request(
    method => 'GET',
    url => 'http://localhost:8080/2legged/',
));

print YAML::Dump($oauth->request(
    method => 'POST',
    url => 'http://localhost:8080/2legged/post/submit',
    params => {id => 'foo', password => 'bar'}
));

